### DISCUSSION TITLE
Support cookie-authenticated httpx client

### DISCUSSION THREAD

**Discussion by davidbrochart** [frankie567/httpx-ws#26] [Q&A]
I am passing a cookie-authenticated `httpx.AsyncClient` to `aconnect_ws`, but it seems that the cookie is not taken into account because I get a 403.
Could it be that this is not implemented yet in `http-ws`?

**Comment by frankie567** [comment]
Normally, `httpx-ws` does nothing about it, it'll pass the headers and cookies "as if" and let HTTPX manage them. Could you try to catch `WebSocketUpgradeError` and log the request headers?

```py
async with httpx.AsyncClient() as client:
    try:
        async with aconnect_ws("http://localhost:8000/ws", client) as ws:
            message = await ws.receive_text()
            print(message)
            await ws.send_text("Hello!")
    except WebSocketUpgradeError as e:
        print(e.response.request.headers)
```

**Reply by davidbrochart** [reply]
Thanks, here are the headers:
```
Headers({'host': '127.0.0.1:36265', 'accept': '*/*', 'accept-encoding': 'gzip, deflate', 'user-agent': 'python-httpx/0.23.3', 'connection': 'upgrade', 'upgrade': 'websocket', 'sec-websocket-key': 'RcrMXNCy0Q1xdtk/ctKkaQ==', 'sec-websocket-version': '13', 'cookie': 'fastapiusersauth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJjNGY3N2NhZC02NmExLTQyZmQtYWRjYS1mZjBlNmI2OWYwOWYiLCJhdWQiOlsiZmFzdGFwaS11c2VyczphdXRoIl19.Fq-_wCdRwKRXnpBf2k6-cu10hUYt13Jt7imE3VCqzRs'})
```
And if I `print(client.cookies)`, I get:
```
<Cookies[<Cookie fastapiusersauth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJjNGY3N2NhZC02NmExLTQyZmQtYWRjYS1mZjBlNmI2OWYwOWYiLCJhdWQiOlsiZmFzdGFwaS11c2VyczphdXRoIl19.Fq-_wCdRwKRXnpBf2k6-cu10hUYt13Jt7imE3VCqzRs for 127.0.0.1 />]>
```

**Reply by frankie567** [reply]
Ok, so cookie is correctly set in the request headers. Are you sure your WebSocket endpoint correctly decodes the token?

**Reply by davidbrochart** [reply]
I figured out the issue, it was just that the WebSocket router was not defined on my side :man_facepalming: 
What was misleading though is that I got a 403, maybe it should have been a 404?
Anyway, thanks a lot for your help, and sorry for the noise!

**Reply by frankie567** [reply]
Ha, great! Well, if you got a 403, I would say you had another router that would take precedence. Otherwise, in standard FastAPI context, you would definitely get a 404.

**Reply by davidbrochart** [reply]
Actually, not for WebSockets: https://github.com/tiangolo/fastapi/issues/572.

**Reply by frankie567** [reply]
Oh wow, I wasn't aware of that, thank you for the pointer! That's quite surprising 😲
